cmd/app.go

```diff
@@ -1080,40 +1082,91 @@ func createNamespace(c *cliclient.MasterClient, n string) error {
 	return nil
 }
 
-// processAnswers adds answers to given map, and prompts users to answers chart questions if interactive is true
+// processAnswers adds answers to map, and prompts users to answers chart questions if interactive is true
+// or uses default answers if interactive is false
 func processAnswers(
 	ctx *cli.Context,
 	c *cliclient.MasterClient,
 	tv *managementClient.TemplateVersion,
 	answers map[string]string,
+	values string,
 	interactive bool,
-) (map[string]string, error) {
+) (map[string]string, string, error) {
 	if answers == nil || ctx.Bool("reset") {
 		// this would not be possible without returning a map
 		answers = make(map[string]string)
 	}
+	// tempAnswers stores answers from the values file. No answers from the values file should be added to the answers map.
+	tempAnswers := make(map[string]string)
+	if ctx.String("values") != "" {
+		// creates an answers map from the values file
+		if err := parseValuesFile(ctx.String("values"), tempAnswers); err != nil {
+			return answers, values, err
+		}
+		var err error
+		// reads the values file to string
+		values, err = parseFileToString(ctx.String("values"))
+		if err != nil {
+			return answers, values, err
+		}
+	}
+	fmt.Println("answersAfterValues: ", answers)
 
-	err := parseAnswersFile(ctx.String("answers"), answers)
-	if err != nil {
-		return answers, err
+	if ctx.String("answers") != "" {
+		err := parseAnswersFile(ctx.String("answers"), answers)
+		if err != nil {
+			return answers, values, err
+		}
 	}
+	fmt.Println("answersAfterAnswers: ", answers)
 
 	for _, answer := range ctx.StringSlice("set") {
 		parts := strings.SplitN(answer, "=", 2)
 		if len(parts) == 2 {
 			answers[parts[0]] = parts[1]
 		}
 	}
-
+	var err error
+	// answers to questions will be added to map
 	if interactive {
-		// answers to questions will be added to map
-		err := askQuestions(tv, answers)
-		if err != nil {
-			return answers, err
+
+		if ctx.String("values") != "" {
+			// need to pass in tempAnswers to ensure questions are not re-asked since answers map is set with
+			// only answers that were not in the file when values file is passed in
+
+			tempAnswers, err = askQuestions(tv, tempAnswers, false)
+			if err != nil {
+				return answers, values, err
+			}
+		} else {
+			answers, err = askQuestions(tv, answers, true)
+			if err != nil {
+				return answers, values, err
+			}
+		}
+	} else {
+		if ctx.String("values") != "" {
+			tempAnswers, err = fillInDefaultAnswers(tv, tempAnswers, false)
+			for k, v := range answers {
+				tempAnswers[k] = v
+			}
```

Comment:
isn't this happening at L1165?

Reply:
yes - I moved it to 1165 but failed to remove it here. good catch! will remove duplicate code